lib/msf/core/exploit/remote/smb/client/kerberos_authentication.rb

```diff
@@ -98,7 +98,12 @@ def smb2_authenticate
     @kerberos_authenticator.validate_response!(response.buffer)
 
     @session_id = response.smb2_header.session_id
-    @application_key = @session_key = kerberos_result[:session_key].value[0...16]
+    if @encryption_algorithm.present?
```

Comment:
If SMB 2 is in use, then `@encryption_algorithm` is nil. This opts to continue using 16 as the default key size unless an encryption algorithm that needs another size is negotiated.